### PR TITLE
Feature/course form fixed

### PR DESCRIPTION
**Pull request by pruthviraj7714**
### PR Fixes:
- 1 #800 
 
### Checklist before requesting a review
- [✔ ] I have performed a self-review of my code
- [ ✔] I assure there is no similar/duplicate pull request regarding same issue

Modified the course add form and used shadcn form with the toaster component and error messages for form validation

before: there is no proper error handling showing and redirection also no validation of course added or not
https://github.com/code100x/cms/assets/138095461/ca5a15db-3360-4c1b-a079-692d6ab72c2c

after:
https://github.com/code100x/cms/assets/138095461/05a3b3a1-0f14-472f-8989-72205348405f

### CODE REVIEW

**Comment by siinghd** [comment]
@pruthviraj7714 whats wrong with 10K+ add? can you update pnpm-lock from main?

**Comment by pruthviraj7714** [comment]
@siinghd i added the component with pnpm

File: package.json [hunk 1]
```diff
@@ -35,8 +35,9 @@
     "@auth/prisma-adapter": "^1.0.6",
     "@discordjs/core": "^1.1.1",
     "@discordjs/next": "^0.1.1-dev.1673526225-a580768.0",
+    "@hookform/resolvers": "^3.6.0",
     "@icons-pack/react-simple-icons": "^9.4.0",
-    "@prisma/client": "^5.6.0",
+    "@prisma/client": "^5.16.1",
```
Comment (siinghd) [inline package.json:40]: why this version change?

File: package.json [hunk 2]
```diff
@@ -116,7 +117,7 @@
     "postcss": "^8",
     "prettier": "^3.2.4",
     "prettier-plugin-tailwindcss": "^0.6.1",
-    "prisma": "^5.13.0",
+    "prisma": "^5.16.1",
```
Comment (siinghd) [inline package.json:120]: same here?

File: package.json [hunk 3]
```diff
@@ -70,7 +71,7 @@
     "pdf-lib": "^1.17.1",
     "react": "^18",
     "react-dom": "^18",
-    "react-hook-form": "^7.50.1",
+    "react-hook-form": "^7.51.0",
```
Comment (siinghd) [inline package.json:74]: also here?

File: src/utiles/appx.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@ const LOCAL_CMS_PROVIDER = process.env.LOCAL_CMS_PROVIDER;
 export async function getPurchases(email: string): Promise<Course[]> {
   const value = await cache.get('courses', [email]);
   if (value) {
-    console.log(value)
+    console.log(value);
```
Comment (siinghd) [inline src/utiles/appx.ts:18]: remove this

**Review by siinghd — CHANGES_REQUESTED** [review]

File: src/utiles/appx.ts (hunk 4 above)
Comment (pruthviraj7714) [reply src/utiles/appx.ts]: @siinghd i don't know why but yarn-lock file showing 12k+ changes

**Comment by pruthviraj7714** [comment]
@siinghd yarn-lock is empty in the main that's why it's showing 12k+ changes

**Pull request merged**